lib/crypto/sr25519/sr25519.go

```diff
@@ -149,9 +157,40 @@ func NewPrivateKey(in []byte) (*PrivateKey, error) {
 	if len(in) != PrivateKeyLength {
 		return nil, errors.New("input to create sr25519 private key is not 32 bytes")
 	}
-	priv := new(PrivateKey)
-	err := priv.Decode(in)
-	return priv, err
+
+	buf := [SeedLength]byte{}
+	copy(buf[:], in)
+	msc, err := sr25519.NewMiniSecretKeyFromRaw(buf)
+	if err != nil {
+		return nil, err
+	}
+
+	priv := msc.ExpandEd25519()
+	return &PrivateKey{key: priv}, nil
+}
```

Comment:
this is duplicate of `NewKeypairFromSeed` - can you use that instead? also, this should be how it was before, since `NewPrivateKey` was intended to create a private key type directly from an input private key